src/sentry_webhook/post_sentry_event_to_salesforce.ts

```diff
@@ -0,0 +1,162 @@
+// an email address.
+export function postSentryEventToSalesforce(
+    event: sentry.SentryEvent, project: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if (!event.user || !event.user.email) {
+      reject(new Error('Not posting event without email'));
+      return;
+    }
+    // Sentry development projects are marked with 'dev', i.e. outline-client-dev.
+    const isProd = project.indexOf('-dev') === -1;
+    const salesforceHost = isProd ? SALESFORCE_PROD_HOST : SALESFORCE_DEV_HOST;
+    const formFields = isProd ? SALESFORCE_FORM_FIELDS_PROD : SALESFORCE_FORM_FIELDS_DEV;
+    const formValues = isProd ? SALESFORCE_FORM_VALUES_PROD : SALESFORCE_FORM_VALUES_DEV;
+    const isClient = project.indexOf('client') !== -1;
+    const formData =
+        getSalesforceFormData(formFields, formValues, event, event.user.email, isClient);
+    const req = https.request(
+        {
+          host: salesforceHost,
+          path: SALESFORCE_PATH,
+          protocol: 'https:',
+          method: 'post',
+          headers: {
+            // The production server will reject requests that do not specify this content type.
+            'Content-Type': 'application/x-www-form-urlencoded'
+          }
+        },
+        (res) => {
+          if (res.statusCode === 200) {
+            resolve();
+          } else {
+            reject(new Error(`Failed to post form data, response status: ${res.statusCode}`));
+          }
+        });
+    req.on('error', (err) => {
+      reject(new Error(
+          `Failed to submit form: ${err}`,
+          ));
+    });
+    req.write(formData);
+    req.end();
+  });
+}
+
+// Returns a URL-encoded string with the Salesforce form data.
+function getSalesforceFormData(
+    formFields: SalesforceFormFields, formValues: SalesforceFormValues, event: sentry.SentryEvent,
+    email: string, isClient: boolean): string {
+  const form = [];
+  form.push(encodeFormData(formFields.orgId, formValues.orgId));
+  form.push(encodeFormData(formFields.recordType, formValues.recordType));
+  form.push(encodeFormData(formFields.email, email));
+  form.push(encodeFormData(formFields.sentryEventId, event.event_id));
+  form.push(encodeFormData(formFields.description, event.message));
+  form.push(encodeFormData(formFields.type, isClient ? 'Outline client' : 'Outline manager'));
+  const tags = getTagsMap(event.tags);
+  if (!!tags) {
+    form.push(encodeFormData(formFields.category, tags.get('category')));
+    form.push(encodeFormData(formFields.os, tags.get('os.name')));
+    form.push(encodeFormData(formFields.version, tags.get('sentry:release')));
+    if (!isClient) {
+      form.push(encodeFormData(formFields.cloudProvider, tags.get('cloudProvider')));
+    }
+  }
+  return form.join('&');
+}
+
+function encodeFormData(field: string, value?: string) {
+  return `${encodeURIComponent(field)}=${encodeURIComponent(value || '')}`;
```

Comment:
FYI `encodeURIComponent` doesn't throw when passed the empty string

Reply:
It encodes `undefined` to `"undefined"` though - rather have an empty string.